swarm/src/lib.rs

```diff
@@ -1901,10 +1914,14 @@ mod tests {
     fn test_swarm_disconnect() {
         // Since the test does not try to open any substreams, we can
         // use the dummy protocols handler.
-        let handler_proto = keep_alive::ConnectionHandler;
+        let handler_proto = dummy::ConnectionHandler;
 
-        let mut swarm1 = new_test_swarm::<_, ()>(handler_proto.clone()).build();
-        let mut swarm2 = new_test_swarm::<_, ()>(handler_proto).build();
+        let mut swarm1 = new_test_swarm::<_, ()>(handler_proto.clone())
```

Comment:
With this feature, I think `new_test_swarm` is always called with `dummy::ConnectionHandler`! This means we can simplify these tests by removing the parameter and constructing `dummy::ConnectionHandler` inside.

Some of the comments around here will have to be updated aswell!